tests/csv_upload_tests.py

```diff
@@ -255,42 +264,101 @@ def test_schema_is_not_allowed(self):
+            assert message in response
+        finally:
+            os.remove(filename)
 
-    # TODO "activate" test when Fields are implemented and code works as intended
+    @unittest.skipIf(
+        "mysql" in conf.get("SQLALCHEMY_DATABASE_URI", "")
+        or "sqlite" in conf.get("SQLALCHEMY_DATABASE_URI", ""),
+        "This test only run when a PostgreSQL database exist",
+    )
     def test_import_into_new_postgres(self):
         url = "/csvimporter/csvtodatabase/add"
         filename = "into_new_postgres.csv"
-        schema = ""
         db_name = "csv_into_new_postgres_db"
         table_name = "newlyimported_into_postgres"
-        # db_flavor = "postgres"
-        # password = "postgres"
-        importer = CsvImporter()
+        form_data = self.get_full_data(
+            filename, -1, db_name, table_name, database_flavor="postgres"
+        )
+        conf["POSTGRES_USERNAME"] = "postgres"
+        conf["POSTGRES_PASSWORD"] = "postgres"
         try:
-            form_data = self.get_full_data(filename, -1, db_name, table_name)
-            # response = self.get_resp(url, data=form_data)
-            # message = "{0} imported into database {1}".format(table_name, db_name)
-            # assert message in response
+            response = self.get_resp(url, data=form_data)
+            message = "{0} imported into database {1}".format(table_name, db_name)
+            print(response)  # show error message in build
+            assert message in response
         finally:
-            assert True
+            os.remove(filename)
+            url = (
+                "postgresql://"
+                + conf["POSTGRES_USERNAME"]
+                + ":"
+                + conf["POSTGRES_PASSWORD"]
+                + "@localhost/"
+                + db_name
+            )
+            if sqlalchemy_utils.database_exists(url):
+                sqlalchemy_utils.drop_database(url)
 
-    # TODO "activate test" and set password to None, assert correct Exception is thrown and caught
-    def test_postgres_no_password_supplied(self):
+    @unittest.skipIf(
+        "mysql" in conf.get("SQLALCHEMY_DATABASE_URI", "")
+        or "sqlite" in conf.get("SQLALCHEMY_DATABASE_URI", ""),
+        "This test only run when a PostgreSQL database exist",
+    )
+    def test_postgres_already_exist(self):
         url = "/csvimporter/csvtodatabase/add"
-        filename = "postgres_no_password.csv"
-        schema = ""
-        db_name = "csv_into_new_postgres_no_pw"
-        table_name = "no_password_supplied"
-        importer = CsvImporter()
-        # db_flavor = "postgres"
+        filename = "postgres_already_exist.csv"
+        db_name = "postgres_already_exist"
+        form_data = self.get_full_data(
+            filename, -1, db_name, database_flavor="postgres"
+        )
+
+        conf["POSTGRES_USERNAME"] = "postgres"
+        conf["POSTGRES_PASSWORD"] = "postgres"
+        CsvImporter()._create_database(db_name, "postgres")
```

Comment:
Couldn't we move this into the try-except block in case we have an unhandled exception? Or is this the reason it's out of the block in order not to shadow those?